# Pre-eclampsia and Eclampsia and Hypertension in Later Life

**Date:** 2021-11-28

**Reference:** M249, Book 1, Part 1

In [1]:
suppressPackageStartupMessages(library(tidyverse))
library(R249)

## Summary

The aim of this notebook was to show how to analyse a cohort study.
Data on a study researching the possible association between hypertension in a woman's first pregnancy with them developing hypertension in later life was taken from a peer-reviewed source.
Measures of association were calculated and a chi-squared test of no association was performed.
It was concluded that the study provided strong evidence of a positive association between hypertension in a woman's first pregnancy with them developing hypertension in later life.

## Get the data

Researchers followed **542** women who suffered from pre-eclampsia or eclampsia during their first pregnancy and **277** women who did not over a period of time, and counted those who developed hypertension in later life and those who did not.
The issue of interest was whether there was evidence of an association between hyptertension in a woman's first pregnancy, and hypertension in later life.

Results from the cohort study are saved in the `data` folder.

In [2]:
(res <- as_tibble(read.csv(file = "..\\..\\data\\preeclampsia.csv")))

count exposure         outcome        
1 327   pre-eclampsia    hypertension   
2 215   pre-eclampsia    no hypertension
3  76   no pre-eclampsia hypertension   
4 201   no pre-eclampsia no hypertension

## Display the data

Cast the `exposure`, `outcome` columns to factors.
This will allow for better sorting of the `tibble`.

In [3]:
labexp <- c("no pre-eclampsia", "pre-eclampsia")
labout <- c("no hypertension", "hypertension")
(sortedres <- res %>%
    mutate(exposure = factor(res$exposure, labexp)) %>%
    mutate(outcome = factor(res$outcome, labout)) %>%
    arrange(exposure, outcome))

count exposure         outcome        
1 201   no pre-eclampsia no hypertension
2  76   no pre-eclampsia hypertension   
3 215   pre-eclampsia    no hypertension
4 327   pre-eclampsia    hypertension

Pull the `count` column as a vector and use it to initialise a matrix.

In [4]:
(resmat <- sortedres %>%
    pull(count) %>%
    matrix(nrow = 2, ncol = 2, byrow = TRUE, dimnames = list(labexp, labout)))

no hypertension hypertension
no pre-eclampsia 201              76         
pre-eclampsia    215             327

## Measures of assocation

Calculate the **odds ratio** and **relative risk**.
The relative risk can be calculated using the `riskratio` function in `R249`.

In [5]:
oddsratio(resmat); riskratio(resmat)

oddsratio stderr    lcb      ucb     
no pre-eclampsia      NA          NA       NA       NA
pre-eclampsia    4.02246   0.1607554 2.935327 5.512225

oddsratio stderr    lcb      ucb    
no pre-eclampsia       NA         NA       NA      NA
pre-eclampsia    2.198946  0.1037346 1.794385 2.69472

The odds ratio is **4.02**, with 95% confidence interval **(2.94, 5.51)**;
the relative risk is **2.20**, with 95% confidence interval **(2.94, 5.51)**.
Both measures are greater than 1 with interval estimates that are strictly greater than **1**.
This suggests there is a positive assosication between hyptertension in the first pregnancy and developing hypertension in later life.

## Strength of evidence of association

Perform a chi-squared test of no association.

In [6]:
chi2 <- chisq.test(resmat, correct = FALSE)

Expected frequencies under a null hypothesis of no association.

In [7]:
chi2$expected

no hypertension hypertension
no pre-eclampsia 140.6984        136.3016    
pre-eclampsia    275.3016        266.6984

The number of women who had hypertension during the first pregnancy and suffered hypertension in later life **(327)** is much greater than expected **(266.70)**.

The null distribution of the test statistic is approximately **chi-squared** with **1** degree of freedom.
Since all expected frequencies are greater than **5**, the approximation is adequate.

Calculate the contribution of each value values to the **chi-squared** test statistic.
We much square the returned matrix as **R** calculates each contribution as

$$
\frac{O-E}{\sqrt{E}}, \hspace{5mm}\text{as opposed to}\hspace{3mm}\frac{(O-E)^{2}}{E}.
$$

In [8]:
(chi2$residuals) ** 2

no hypertension hypertension
no pre-eclampsia 25.84451        26.67820    
pre-eclampsia    13.20836        13.63443

Return the test statistic and **p**-value.

In [9]:
chi2


	Pearson's Chi-squared test

data:  resmat
X-squared = 79.366, df = 1, p-value < 2.2e-16


The test statistic is approximately **79.633**, with **p**-value approximately **0.0000**.
Given **p < 0.000**, there is very strong evidence against the null hypothesis.
Hence, there is strong evidence of a positive associaition between hypertension during the first pregnancy and hypertension in later life.